### PR TITLE
Kuz362 internal index security

### PR DESCRIPTION
**Pull request by scottinet**
- added a new `internalIndex` global configuration
- replaced all explicit references to the `%kuzzle` index with the new `internalIndex` configuration
- index wildcards now resolve all indexes except the internal one. Rights on the internal index must now be explicitly granted
- security bugfix: the `security` controller cannot be invoked anymore if no explicit rights have been granted on the internal index 
- added explicit rights on the internal index to the default `guest` and `admin` roles
- role and profile creation now use the user's context instead of a generic one

### CODE REVIEW

File: lib/api/core/models/repositories/roleRepository.js [hunk 1]
```diff
@@ -173,7 +173,7 @@ module.exports = function (kuzzle) {
    */
   RoleRepository.prototype.validateAndSaveRole = function (role) {
```
Comment (stafyniaksacha) [inline lib/api/core/models/repositories/roleRepository.js]: Why we don't give the real context there ?

File: lib/api/core/models/repositories/roleRepository.js [hunk 2]
```diff
@@ -173,7 +173,7 @@ module.exports = function (kuzzle) {
...
     var context = {
-      connection: {type: 'websocket'},
+      connection: {type: 'internal'},
```
Comment (stafyniaksacha) [inline lib/api/core/models/repositories/roleRepository.js]: Why we don't give the real context there ?

Comment (scottinet) [inline lib/api/core/models/repositories/roleRepository.js]: I was afraid of potential side-effects, and too lazy to investigate further :stuck_out_tongue: 

It seems this function is only called by the security controller, so I'll add the real context to this one, and to other similar functions used by profiles and users management.

**Comment by jenow** [comment]
+1

**Comment by stafyniaksacha** [comment]
+1

**Comment by ballinette** [comment]
+1

**Comment by scottinet** [comment]
Got 3 approvals => merging

**Pull request merged**